addons/web_editor/models/field_html_history_mixin.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+from .patch_utils import apply_patch, generate_comparison, generate_patch
+
+
+class HtmlHistory(models.AbstractModel):
+    _name = "field.html.history.mixin"
+    _description = "Field html History"
+
+    history_revision_ids = fields.One2many(
+        "field.html.history.revision", "res_id", string="Related revision Ids",
+        copy=True)
+
+    def _get_versioned_field_names(self):
+        """ This method should be overriden
+
+            :return: list[string]: The names of the fields to be versioned
+        """
+        return []
+
+    def write(self, vals):
+        versioned_fields = self._get_versioned_field_names()
+        for field_name in versioned_fields:
+            if field_name not in vals:
+                continue
+
+            new_content = vals[field_name]
+            if self.id and isinstance(new_content, str):
+                old_content = getattr(self, field_name)
+                if isinstance(old_content, str) and new_content != old_content:
+                    patch = generate_patch(new_content, old_content)
+                    self.env['field.html.history.revision'].create({
+                        'res_id': self.id,
+                        'res_model': self._name,
+                        'res_field': field_name,
+                        'patch': patch})
+
+        return super().write(vals)
+
+    def unlink(self):
+        """ Delete all HtmlHistoryDiff related to this document """
+        self.env['field.html.history.revision'].search(
+            [('res_id', 'in', self.ids)]).unlink()
+        return super().unlink()
+
+    def get_revisions_for_field(self, field_name):
+        """ Get the list of revision linked to the provided field name.
+            Order from the most recent to the oldest.
+
+            :param str field_name: the name of the field
+
+            :return: list: list of revision
+        """
+        self.ensure_one()
+        return self.env['field.html.history.revision'].search(
+            [('res_id', '=', self.id),
+             ('res_model', '=', self._name),
+             ('res_field', '=', field_name)], order='id desc')
```

Comment:
Isn't it simpler using fields, like
```suggestion
        return self.history_revision_ids.filtered(lambda revision: revision.res_field == field_name)
```